glue/qt/widgets/histogram_widget.py

```diff
@@ -193,6 +183,21 @@ def component(self, component):
 
     @defer_draw
     def _set_attribute_from_combo(self, *args):
+        if self.component is not None:
+            for d in self._data:
+                try:
+                    component = d.get_component(self.component)
+                except:
+                    continue
+                else:
+                    break
```

Comment:
Hmm, I guess that one can end up with a component ID referring to different data:

```python
In [2]: d1 = Data(x=[1,2,3])

In [3]: d2 = Data(y=[4,5,6])

In [4]: c1 = d1.find_component_id('x')

In [5]: c2 = d2.find_component_id('y')

In [6]: 

In [6]: d1.update_id(c1, c2)

In [7]: 

In [7]: print(d1[c2])
[1 2 3]

In [8]: print(d2[c2])
[4 5 6]
```